site/app/controllers/forum/ForumController.php

```diff
@@ -557,8 +604,13 @@ public function getSinglePost() {
      */
     public function alterAnnouncement(bool $type) {
         $thread_id = $_POST["thread_id"];
+        if ($type == 'announce_thread') {
```

Comment:
It's always a good idea to use strict equality in PHP.  Apply throughout.
```suggestion
        if ($type === 'announce_thread') {
```